src/plots/cartesian/axes.js

```diff
@@ -1847,7 +1774,7 @@ axes.doTicks = function(gd, axid, skipTitle) {
     if(axLetter === 'x') {
         sides = ['bottom', 'top'];
         transfn = function(d) {
-            return 'translate(' + ax.l2p(d.x) + ',0)';
+            return 'translate(' + (ax._offset + ax.l2p(d.x)) + ',0)';
```

Comment:
At some point I think we'd be better off just merging `ax._offset` into `ax.*2p` and getting rid of all the `<g>` translations we have scattered about. The fact that it's not in there is a remnant of the days when plots were nested `<svg>` elements so within a subplot pixel coordinates always started at (0,0). Here at least I'm getting rid of the `<g>` translations associated with ticks and grids, but it means I have to put them back into the tick and grid *paths* until `*2p` catches up.